docs/canonicalk8s/snap/explanation/roles.md

```diff
@@ -0,0 +1,38 @@
+# Node roles
+
+This document explains how {{product}} assigns Kubernetes node roles, how this
+affects workload scheduling, and how this differs from standard Kubernetes
+implementations.
+
+When bootstrapping a cluster, {{product}} assigns the following roles to nodes:
+
+| Node Type | Default Roles | Scheduling | Notes |
+|-----------|--------------|------------|-------|
+| Control Plane | `control-plane`, `worker` | Allowed | Differs from `kubeadm` which prevents workload scheduling on control plane nodes |
+| Worker | `worker` | Allowed | Standard behavior |
+
+## Rationale
+
+Most kubernetes implementations (like kubeadm) set a NoSchedule taint on
+control plane nodes as a security measure to isolate control planes from
+workloads.
+
+{{product}} intentionally allows workload scheduling on control plane nodes to
+simplify initial cluster setup, especially for single-node deployments.
+However, users should be aware that:
+
+- Scheduling workloads on control plane nodes may introduce security risks
+- For production multi-node clusters, isolating the control plane is
+  recommended
+
+If you decide to deploy a multi-node cluster, you can set a taint on the
+control plane node using this command:
+
+## Modifying default behaviour
```

Comment:
I think there could be some confusion here between default behavior of upstream K8s and Canonical K8s. I would maybe change the heading to be something along the lines of Modifying the node roles?